OptimizelySDK/OptlyConfig/OptimizelyConfigService.cs

```diff
@@ -154,12 +215,121 @@ public OptimizelyConfigService(ProjectConfig projectConfig)
         /// <param name="projectConfig">The project config</param>
         /// <returns>Dictionary | Dictionary of FeatureFlag key and value as list of all FeatureVariable inside it</returns>
         private IDictionary<string, List<FeatureVariable>> GetFeatureIdVariablesMap(ProjectConfig projectConfig)
-        {            
+        {
             var featureIdVariablesMap = projectConfig?.FeatureFlags?.ToDictionary(k => k.Id, v => v.Variables);
-            
+
             return featureIdVariablesMap ?? new Dictionary<string, List<FeatureVariable>>();
         }
 
+        /// <summary>
+        /// Gets stringify audiences used in given experiment
+        /// </summary>
+        /// <param name="experiment">The experiment</param>
+        /// <param name="projectConfig">The project config</param>
+        /// <returns>string | Audiences used in experiment.</returns>
+        private string GetExperimentAudiences(Experiment experiment, ProjectConfig projectConfig)
+        {
+            if (experiment.AudienceConditionsString == null)
+            {
+                return "";
+            }
+            var s = JsonConvert.DeserializeObject<List<object>>(experiment.AudienceConditionsString);
+            return GetSerializedAudiences(s, projectConfig.AudienceIdMap);
+        }
+
+        /// <summary>
+        /// Converts list of audience conditions to serialized audiences used in experiment
+        /// </summary>
+        /// <param name="audienceConditions">List of audience conditions in experiment</param>
+        /// <param name="audienceIdMap">The audience Id map</param>
+        /// <returns>string | Serialized audience in which IDs are replaced with audience name.</returns>
+        private string GetSerializedAudiences(List<object> audienceConditions, Dictionary<string, Audience> audienceIdMap)
```

Comment:
use stringbuilder in this class, need to see again.

Reply:
to better understand, can you please add some examples in the comments. would be easy to understand the logics.